### PR TITLE
[feat] make error widget sexy

### PR DESCRIPTION
**Pull request by a-wallen**
### Feature Preview

image

This is to style the error widget per @appflowy's request and fix my remaining TODO comment.

---

#### PR Checklist

- [X] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [ ] ~I've listed at least one issue this PR fixes in the description above.~ fixing TODO
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/packages/flowy_infra_ui/lib/widget/error_page.dart [hunk 1]
```diff
@@ -1,11 +1,201 @@
...
+    Key? key,
+  }) =>
+      FlowyErrorPage._(
+        e.toString(),
+        stackTrace: e.stackTrace?.toString(),
+        howToFix: howToFix,
+        key: key,
+      );
+
+  factory FlowyErrorPage.error(
```
Comment (a-wallen) [inline frontend/appflowy_flutter/packages/flowy_infra_ui/lib/widget/error_page.dart:8]: I wanted to make a constructor to get the `FlowyErrorPage` from a `FlowyError`, but sadly it isn't available for `flowy_infra_ui`.

Comment (appflowy) [reply frontend/appflowy_flutter/packages/flowy_infra_ui/lib/widget/error_page.dart:8]: Do you mean we can't reference to the `FlowyError`? If yes, how about adding the deps?

File: frontend/rust-lib/dart-ffi/Cargo.toml [hunk 2]
```diff
@@ -7,8 +7,8 @@ edition = "2018"
 [lib]
 name = "dart_ffi"
 # this value will change depending on the target os
-# default staticlib
-crate-type = ["staticlib"]
+# default cdylib
```
Comment (appflowy) [inline frontend/rust-lib/dart-ffi/Cargo.toml:10]: This file should't be modified. So you can ignore this change

**Comment by appflowy** [comment]
When an error occurs, I want to prompt the user to take some action. Perhaps provide some buttons and ask the user to tap one of them. How can I setup in this page?

File: frontend/appflowy_flutter/packages/flowy_infra_ui/lib/widget/error_page.dart (hunk 1 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/packages/flowy_infra_ui/lib/widget/error_page.dart:8]: Yes, `FlowyError` is generated and available in the backend package. I'll try to add the dependency and get back to you.

**Comment by a-wallen** [comment]
> When an error occurs, I want to prompt the user to take some action. Perhaps provide some buttons and ask the user to tap one of them. How can I setup in this page?

The `FlowyErrorPage` requires that `howToFix` is provided. The fallback instructions are to click the GitHub button below to submit a bug on our issues page. Do you want me to add an optional field for actions? I think that you might want an API similar to `AlertDialog`, but I might be mistaken.

image

Let me know if that's what you mean, it should be an easy patch.

**Comment by appflowy** [comment]
> > When an error occurs, I want to prompt the user to take some action. Perhaps provide some buttons and ask the user to tap one of them. How can I setup in this page?
> 
> The `FlowyErrorPage` requires that `howToFix` is provided. The fallback instructions are to click the GitHub button below to submit a bug on our issues page. Do you want me to add an optional field for actions? I think that you might want an API similar to `AlertDialog`, but I might be mistaken.
> 
> image
> 
> Let me know if that's what you mean, it should be an easy patch.


Some errors need to be presented to the user, and we should wait for their feedback. For example, if the database file is corrupted, I want to display a restore button on the FlowyErrorPage. If this feature is currently supported, that would be great.

**Pull request merged**